include/boost/geometry/algorithms/clear.hpp

```diff
@@ -90,7 +91,7 @@ namespace dispatch
 template
 <
     typename Geometry,
-    typename Tag = typename tag_cast<typename tag<Geometry>::type, multi_tag>::type
+    typename Tag = tag_cast_t<tag_t<Geometry>, multi_tag>
```

Comment:
I'll finish my review but this all looks very good, cool improvement. Way more readable.

Thanks for all this work.